src/components/form/file-upload/image-upload.tsx

```diff
@@ -0,0 +1,169 @@
+import { ComponentPropsWithRef, forwardRef, useRef, useState } from 'react';
+import { Hint, Paragraph } from '../../..';
+import { cn } from '../../../utils/tailwind';
+import { Icon } from '../../icon';
+
+export interface IImageUpload
+  extends Omit<ComponentPropsWithRef<'input'>, 'type' | 'className' | 'accept' | 'aria-hidden' | 'label'> {
+  id: string;
+  name: string;
+  error?: string;
+  disabled?: boolean;
+}
+
+export const ImageUpload = forwardRef<HTMLInputElement, IImageUpload>(
+  ({ name, error, disabled, id, onChange, ...rest }, ref) => {
+    const labelRef = useRef<HTMLLabelElement>(null);
```

Comment:
Vielleicht wir sollten alle states mergen in ein einzige State? ich habe mal gelesen, multiple states nacheinander ist schlecht.
Zum Beispiel const [state, setsState] = useState({
  dragActive: '',
  fileName: '',
  fileSize: ''
  ...
})

Dann kann man updaten:

setState({
  ...state,
  fileName: 'test234'  
})

Reply:
cool, die idee gefällt mir 👍 hab es angepasst